rfcs/rdm-0064-custom-fields.md

```diff
@@ -0,0 +1,559 @@
+            "icon": "pencil",
+            "description": _(
+                "You should fill this field with one of the experiments e.g LHC, ATLAS etc.",
+            )
+        },
+    }]
+}]
+```
+
+The instance administrator is defining a list of sections, with each section defining its own fields respectively. By using these sections, they can semantically group together different fields. The order in which sections and fields are configured is respected in the UI. All the sections and fields will be appended in the end of the configured form i.e upload form and community profile page.
+
+Each field can be configured with following parameters:
+
+| Config property |  Type  |                    Description                     |
+| :-------------: | :----: | :------------------------------------------------: |
+|      field      | string | Should be set to one of the defined custom fields. |
+|    ui_widget    | string |           Name of the UI widget to use.            |
+|      props      |  dict  |   Object of properties passed to the `ui_widget`   |
+
+An example of `props` passed in the example configuration above is analyzed below:
+
+| Widget property |  Type  |                          Description                          |
+| :-------------: | :----: | :-----------------------------------------------------------: |
+|      label      | string |              UI label to display for the field.               |
+|   placeholder   | string |                  Placeholder for user input.                  |
+|      icon       | string |                    Semantic UI icon name.                     |
+|   description   | string | Description to be used as helptext for the new defined field. |
+
+Moreover, any other property can be passed in the respective UI widget via the `props` key.
+
+#### Adding custom fields to the forms
+
+> This section focuses on adding/overriding widgets for the records upload or communities settings form. How to display them on existing records is explained in the following section.
+
+In the previous section, the `ui_widget` field was mentioned. Several of them
+are already available with InvenioRDM, however, in some cases it might be
+desired to change their behaviour (override) or use a totally new widget.
+
+UI Widget are discovered and imported in the following way:
+
+1. Try to import the widget from the `my-site` instance, looking in the `/assets/templates` folder (that folder is copied/symlinked when the assets are built).
+2. If the widget is not found, then it tries to import it from the local module's path, e.g. `invenio-communities`.
+3. If the widget is still not found, then it raises an error.
+
+With the above mechanism, instance administrators can add new files/UI widgets that can then be referenced in the config. For example, if an instance administrator wants to add a new UI widget called `MyCustomWidget`, then they need to:
+
+1. Go to `/path/to/my-site/assets/templates/custom_fields/` folder
```

Comment:
For the code, do we use `dash` or `underscore`?
```suggestion
1. Go to `/path/to/my-site/assets/templates/custom-fields/` folder
```

Reply:
It's a dash!